pyproject.toml

```diff
@@ -1,41 +1,159 @@
+]
+dependencies = [
+    "jupyter_client>=6.1.3,<8",
+    "jupyter_core>=4.11.0",
+    "jupyter_server>=1.18,<2.0.0",
+    "jupyterlab_server>=2.3.0,<3",
+    "nbclient>=0.4.0,<0.6",
+    "nbconvert>=6.4.5,<7",
+    "traitlets>=5.0.3,<6",
+    "websockets>=9.0",
+]
+dynamic = [
+    "version",
+]
+
+[project.license]
+file = "LICENSE"
 
-[license]
-file="LICENSE"
+[project.optional-dependencies]
+dev = [
+    "black",
+    "hatch",
+    "jupyter_releaser~=0.25",
+]
+test = [
+    "ipywidgets",
+    "matplotlib",
+    "mock",
+    "numpy",
+    "pandas",
+    "papermill",
+    "pytest",
+    "pytest-rerunfailures",
+    "pytest-tornasync",
+]
+
+[project.scripts]
+voila = "voila.app:main"
+
+[project.urls]
+Homepage = "https://github.com/voila-dashboards/voila"
 
-[tool.jupyter-packaging.options]
-skip-if-exists = ["voila/labextension/static/style.js"]
-ensured-targets = ["voila/labextension/static/style.js"]
+[tool.hatch.version]
+path = "voila/_version.py"
 
-[tool.jupyter-packaging.builder]
-factory = "jupyter_packaging.npm_builder"
+[tool.hatch.build]
+artifacts = [
+    "voila/labextension",
+    "share/jupyter/voila/templates"
+]
+
+[tool.hatch.build.hooks.custom]
+path = "hatch_build.py"
 
-[tool.jupyter-packaging.build-args]
+[tool.hatch.build.targets.wheel.shared-data]
+"etc/jupyter" = "etc/jupyter"
+"voila/static" = "share/jupyter/nbextensions/voila"
+"voila/labextension" = "share/jupyter/labextensions/@voila-dashboards/jupyterlab-preview"
+"install.json" = "share/jupyter/labextensions/@voila-dashboards/jupyterlab-preview/install.json"
+"share/jupyter/voila/templates" = "share/jupyter/voila/templates"
+
+[tool.hatch.build.targets.sdist]
+exclude = [
+    ".github",
+    "*.gif",    
+]
+
+[tool.hatch.build.hooks.jupyter-builder]
+dependencies = [
+    "hatch-jupyter-builder>=0.5.3",
+]
+build-function = "hatch_jupyter_builder.npm_builder"
+ensured-targets = [
+    "voila/labextension/static/style.js",
+    "share/jupyter/voila/templates/classic/static/materialcolors.css",
+    "share/jupyter/voila/templates/classic/static/labvariables.css",
+    "share/jupyter/voila/templates/reveal/static/materialcolors.css",
+    "share/jupyter/voila/templates/reveal/static/labvariables.css",
+]
+skip-if-exists = [
+    "voila/labextension/static/style.js",
+    "share/jupyter/voila/templates/classic/static/materialcolors.css",
+    "share/jupyter/voila/templates/classic/static/labvariables.css",
+    "share/jupyter/voila/templates/reveal/static/materialcolors.css",
+    "share/jupyter/voila/templates/reveal/static/labvariables.css",
+]
+
+[tool.hatch.build.hooks.jupyter-builder.editable-build-kwargs]
+build_cmd = "build"
+npm = [
+    "jlpm",
+]
+force = true
+
+[tool.hatch.build.hooks.jupyter-builder.build-kwargs]
 build_cmd = "build:prod"
-npm = ["jlpm"]
+npm = [
+    "jlpm",
+]
+
+# This is only so that we can pin to an ipyvolume branch, we should remove this when we can
+[tool.hatch.metadata]
+allow-direct-references = true
 
```

Comment:
This can probably be removed now that the ipyvolume dep is in a separate file

```suggestion
```

Reply:
Good point.